spec/projection_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'check_graphite/projection'
+require 'nagios_check'
+
+describe describe CheckGraphite::Projection do
+  let :check do
+    Class.new do
+      attr_reader :values
+      include NagiosCheck
+      include CheckGraphite::Projection
+    end.new
+  end
+
+  describe 'primary value' do
+    subject do
+      check.prepare
+      check.send(:parse_options, ['--projection', projection])
+      check.options.processor.call(datapoints)
+      check.values.first[1]
+    end
+
+    context 'given a linearly decreasing series and a projection of 2 sec' do
+      let(:projection) { '2sec' }
+      let(:datapoints) { [[10,0], [9,1], [8,2]] }
+
+      it { should be_within(0.01).of(6) }
+    end
+
+    context 'given a constant series and a projection of 2 sec' do
+      let(:projection) { '2sec' }
+      let(:datapoints) { [[10,0], [10,1], [10,2]] }
+
+      it { should be_within(0.01).of(10) }
+    end
+  end
+
+  describe 'p-value' do
+    subject do
+      check.prepare
+      check.send(:parse_options, ['--projection', projection])
+      check.options.processor.call(datapoints)
+      check.values['p-value']
+    end
+
+    context 'given a constant series (where Pearson is undefined)' do
+      let(:projection) { '2sec' }
+      let(:datapoints) { [[10,0], [10,1], [10,2]] }
+      it { should eq('undefined') }
+    end
+
+    context 'given a linearly decreasing series' do
+      let(:projection) { '2sec' }
+      let(:datapoints) { [[10,0], [9,1], [8,2]] }
+
+      it { should eq("1.0") }
+    end
+  end
+
+  describe 'integration test' do
```

Comment:
The main problem here is that the test resides inside CheckGraphite::Projection at all. Rather, it is a test to make sure that projections are invoked. Even asserting on the output is a bit overreach. Changing this to a separate describe saying `when invoking graphite with --projection`.